docs/process/guidance/feature_flags/index.rst

```diff
@@ -0,0 +1,507 @@
+
+Feature flags of this category reside in
+`eclipse-score/score:flags`_.
+
+Configuration of a feature
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To modify behavior of a feature, the documentation in the feature tree may point out several feature flags. Such flags
+are of this category.
+
+The name of the flag is defined as `<feature_name>_<configuration>` where:
+
+- `<feature_name>` corresponds to the name of the feature in `snake_case`
+- `<configuration>` corresponds to the name of the configuration in `snake_case`
+
+Feature flags of this category reside in
+`eclipse-score/score:flags`_.
+
+Implementation specific configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Modules may have configuration options that are implementation specific. Such options are not mentioned in the feature
+tree but inside the documentation of the module itself.
+
+The name of the flag is defined as `<feature_name>_<module_name>_<configuration>` where:
+
+- `<feature_name>` corresponds to the name of the feature in `snake_case`
+- `<module_name>` corresponds to the name of the module in `snake_case`
+- `<configuration>` corresponds to the name of the configuration in `snake_case`
+
+Feature flags of this category reside in a top-level directory called `flags` of the module.
+
+Propagation of feature flags
+----------------------------
+
+To take effect beyond Bazel, the values of features needs to be propagated to
+different programming languages.
+
+Definition of Public APIs
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: bazel
+    :caption: some_module/BUILD
+
+    alias(
+        name = "public_lib",
+        actual = select({
+            ":config_some_feature": [":lib"],
+            "//conditions:default": [],
+        }),
+    )
+
+    cc_library(
+        name = "lib",
+        # ...
+    )
+
+Please refrain from trying to use the approach below.
+It will not work, since `select()` cannot be combined with `visibility`.
+
+.. code-block:: bazel
+    :caption: some_module/BUILD
+
+    cc_library(
+        name = "public_lib",
+        visibility = select({
+          ":config_some_feature": ["//visibility:__public__"],
+          "//conditions:default": [],
+        }),
+    )
+
+C++
+^^^
+
+The selection which tests shall be executed depending on a feature values shall
+be done via preprocessor macros in the test.cpp
+
+The activation/deactivation of features is propagated via `local_defines`.
+
+.. code-block:: cpp
+    :caption: some_module/test.cpp
+
+    #if defined(SOME_FEATURE)
+        EXPECT_CALL(...)
+    #else
+        EXPECT_CALL(...)
+
+.. code-block:: bazel
+    :caption: some_module/BUILD
+
+    cc_library(
+        name = "lib",
+        local_defines = select({
+            ":config_some_feature": ["SOME_FEATURE"],
+            "//conditions:default": [],
+        }),
+        # ...
+    )
+
+Python
+^^^^^^
+
+We recommend to propagate features via command line arguments. e.g.
```

Comment:
Can you mention what this will result in?
`foo.py` vs `foo.py SOME_FEATURE`?

How about using something more typical like `--some_feature`?

Reply:
Fine for me, will change.